oracle/src/services/amqpClient.js

```diff
@@ -24,21 +24,39 @@ function connectWatcherToQueue({ queueName, cb }) {
 }
 
 function connectSenderToQueue({ queueName, cb }) {
+  const deadLetterExchange = `${queueName}-retry`
+
   const channelWrapper = connection.createChannel({
     json: true
   })
 
   channelWrapper.addSetup(channel => {
     return Promise.all([
+      channel.assertExchange(deadLetterExchange, 'fanout', { durable: true }),
       channel.assertQueue(queueName, { durable: true }),
+      channel.bindQueue(queueName, deadLetterExchange),
       channel.prefetch(1),
       channel.consume(queueName, msg =>
         cb({
           msg,
           channel: channelWrapper,
           ackMsg: job => channelWrapper.ack(job),
           nackMsg: job => channelWrapper.nack(job, false, true),
-          sendToQueue: data => channelWrapper.sendToQueue(queueName, data, { persistent: true })
+          scheduleForRetry: async (data, msgRetries = 0) => {
+            const retries = msgRetries + 1
+            const delay = retries ** 2 * 1000
```

Comment:
we should limit this I think. as so it would not be too huge. E.g. 60 second between attempts is enough.

I would also suggest usage of Fibonacci sequence: 1, 2, 3, 5, 8, 13, 21, 34 or something like this. It will make short periods between first attempts (by hope that the situation is solved quickly) and provide reasonable delays if the situations was not fixed quickly.

Reply:
Done https://github.com/poanetwork/tokenbridge/pull/103/commits/a466fe57dcfc9feb95acbf3421ece11f4980a5c6